### DISCUSSION TITLE
Adding mirror.linkerd.io/exported=true label doesn't immediately cause service-mirror to mirror the service

### DISCUSSION THREAD

**Discussion by NeckBeardPrince** [linkerd/linkerd2#8469] [Help] [answered]
After adding the `mirror.linkerd.io/exported=true` label to a service in cluster A., Why doesn't the service-mirror in cluster B create the mirrored service until an endpoint is changed in the source service?

I have a headless Elasticsearch service in cluster A. When I added the `mirror.linkerd.io/exported=true` label to that service, I can see the service-mirror in cluster B noticed the change. `OnUpdateCalled: {svc: Service: {name: rockwell-es-ingest, namespace: observability, annotations: [[]], labels [[]]}}","time":"2022-05-11T00:35:32Z"}` and `RemoteServiceCreated: {service: Service: {name: rockwell-es-ingest, namespace: observability, annotations: [[]], labels [[]]}}","time":"2022-05-11T00:35:32Z"}`. But the mirrored service doesn't show up in cluster B until I either restart the pod in cluster A or do something else that would cause an endpoint change. When I restart a pod or do something else that will cause an endpoint change, it picks it up.

`"Received: \u0026{%!s(*v1.Endpoints=\u0026{{ } {rockwell-es-ingest  observability  cece188e-32ad-4df8-90c6-d4c79ff7e24b 235349 0 {{0 63787819392 0x2e52340}} \u003cnil\u003e \u003cnil\u003e map[common.k8s.elastic.co/type:elasticsearch elasticsearch.k8s.elastic.co/cluster-name:rockwell elasticsearch.k8s.elastic.co/statefulset-name:rockwell-es-ingest mirror.linkerd.io/exported:true service.kubernetes.io/headless:] map[] [] []  [{kube-controller-manager Update v1 0xc0004f0db0 FieldsV1 0xc0004f0dc8}]} [{[{10.3.0.11 rockwell-es-ingest-0 0xc0005d9b40 0xc0001f65b0} {10.3.1.24 rockwell-es-ingest-1 0xc0005d9b50 0xc0001f6620}] [] [{http 9200 TCP \u003cnil\u003e}]}]})}","time":"2022-05-11T00:47:45Z"}`

**Comment by alpeb** [comment, marked as answer]
Thanks for the report @NeckBeardPrince . I can reproduce the bug; I'll check back when a fix is implemented.

**Reply by NeckBeardPrince** [reply]
@alpeb I had a chat with @mateiidavid on this issue in Slack. [Here](https://linkerd.slack.com/archives/C89RTCWJF/p1652230311882289) is a link to the thread, if it can give you any more details.

**Reply by NeckBeardPrince** [reply]
@alpeb Is there a chance this is due to it being a mirroed headless service?

**Reply by alpeb** [reply]
Yep, it's a bug specific to mirrored headless services. I've got a fix up for review at #8475

**Reply by NeckBeardPrince** [reply]
@alpeb Awesome. Thank you!
